code/drasil-code/lib/Language/Drasil/Code/Imperative/ReadInput.hs

```diff
@@ -83,7 +83,7 @@ strAsExpr _        _ = error "strAsExpr should only be numeric space or string"
 
 -- | Gets the dimension of a 'Space'.
 getDimension :: Space -> Int
-getDimension (Vect t) = 1 + getDimension t
+getDimension (ClifS _ _ s) = 1 + getDimension s -- TODO: Does this make sense? Maybe we're overloading the term "dimension" now.
```

Comment:
I agree, this is weird. Can you figure out who calls `getDimension` and what they need from it?  It looks like this is always going to return '1' for a `Vector` and 0 in all other cases!

Reply:
Well, we had things like `Vect (Vect Natural)` which would return 2.

Reply:
Ah, yes. Horrible hack that. See if you can get rid of it?